backend/src/server_process_helper.py

```diff
@@ -47,21 +47,28 @@ def __init__(self, flags: list[str]):
         self._stop_event = threading.Event()
 
         # Create a separate thread to read and print the output of the subprocess
-        threading.Thread(
+        self.t1 = threading.Thread(
             target=self._read_output,
             daemon=True,
             name="output reader",
-        ).start()
+        )
+        self.t1.daemon = True
+        self.t1.start()
 
     def close(self):
+        logger.info("Closing worker process...")
         self._stop_event.set()
-        self._process.terminate()
-        self._process.kill()
+        if hasattr(self, "_process"):
+            self._process.terminate()
+            self._process.kill()
+            del self._process
+        if hasattr(self, "t1"):
+            del self.t1
```

Comment:
Don't do that... Please just set it to `None`, so pyright can do its thing.

Reply:
Set it to none where? I don't even know what was causing it to not be initialized in the first place (look at the failing CI commits)

Reply:
Sorry for being dumb lol